### ISSUE TITLE
Reflection warnings and deprecated methods usage in shape.clj

### ISSUE THREAD

**Issue by jsyrjala** [cognitect-labs/aws-api#106]
## Dependencies

``` clojure
[org.clojure/clojure "1.10.1"]
[com.cognitect.aws/api "0.8.391"],
[com.cognitect.aws/endpoints "1.1.11.670"]
[com.cognitect.aws/s3 "770.2.568.0"]
[com.cognitect.aws/lambda "770.2.568.0"]
```

## Description with failing test case

There are reflection warnings when doing `lein uberjar` with

```clojure
  :profiles {:uberjar {:aot :all
                       :global-vars {*warn-on-reflection* true}
                       }
```

```
Reflection warning, cognitect/aws/shape.clj:85:11 - call to java.util.Date ctor can't be resolved.
Reflection warning, cognitect/aws/shape.clj:89:11 - call to java.util.Date ctor can't be resolved.
Reflection warning, cognitect/aws/shape.clj:397:45 - call to java.lang.Double ctor can't be resolved.
Reflection warning, cognitect/aws/shape.clj:398:45 - call to java.lang.Double ctor can't be resolved.
Reflection warning, cognitect/aws/shape.clj:399:45 - call to java.lang.Long ctor can't be resolved.
Reflection warning, cognitect/aws/shape.clj:400:45 - call to java.lang.Long ctor can't be resolved.
```

`java.util.Date` warnings are about this:
```clojure
(defn parse-date
  [shape data]
  (condp = (:timestampFormat shape)
    "rfc822"  (util/parse-date util/rfc822-date-format data)
    "iso8601" (util/parse-date util/iso8601-date-format data)
    (cond (int? data)
          (java.util.Date. (* 1000 data))
          (double? data)
          (java.util.Date. (* 1000 (long data)))
          (re-matches #"^\d+$" data)
          (java.util.Date. (* 1000 (read-string data)))
          :else
          (->> [util/iso8601-date-format
                util/iso8601-msecs-date-format
                util/rfc822-date-format]
               (map #(try (util/parse-date % data) (catch java.text.ParseException _ nil)))
               (filter identity)
               first))))
```

`java.lang.Double` and `java.lang.Long` warnings are about these:
```clojure
(defmethod xml-parse* "double"    [_ nodes] (Double. (data nodes)))
(defmethod xml-parse* "float"     [_ nodes] (Double. (data nodes)))
(defmethod xml-parse* "long"      [_ nodes] (Long. (data nodes)))
(defmethod xml-parse* "integer"   [_ nodes] (Long. (data nodes)))
```

Also this last snippet uses 
[java.lang.Double](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Double.html#%3Cinit%3E(double)) and [java.lang.Long](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Long.html#%3Cinit%3E(long)) constructors. They both are marked as deprecated since JDK 9, and JavaDocs recommend using `Double/valueOf`  and `Long/valueOf` instead.

**Comment by dchelimsky** [comment]
I'd already dealt with some of these [here](https://github.com/cognitect-labs/aws-api/commit/49b7d218f65844ad50ae59a7b1f71ca1f369425f). Just fixed more [here](https://github.com/cognitect-labs/aws-api/commit/2bbce108ec8d3f0a749c014989e8411049b9c665). I think I've got them all now. Thanks!

**Comment by dchelimsky** [comment]
Fix released in 0.8.408
